cpp/src/parquet/encoder.cc

```diff
@@ -79,6 +79,15 @@ class EncoderImpl : virtual public Encoder {
 
   MemoryPool* memory_pool() const override { return pool_; }
 
+  int64_t ReportUnencodedDataBytes() override {
+    if (descr_->physical_type() != Type::BYTE_ARRAY) {
+      throw ParquetException("ReportUnencodedDataBytes is only supported for BYTE_ARRAY");
+    }
+    int64_t byte = unencoded_byte_array_data_bytes_;
+    unencoded_byte_array_data_bytes_ = 0;
+    return byte;
```

Comment:
Nit: let's make the name less confusing?
```suggestion
    int64_t bytes = unencoded_byte_array_data_bytes_;
    unencoded_byte_array_data_bytes_ = 0;
    return bytes;
```